mirdata/datasets/saraga_hindustani.py

```diff
@@ -0,0 +1,499 @@
+
+    tempo_annotation = {}
+    head, tail = os.path.split(tempo_path)
+    sections_path = tail.split('.')[0] + '.sections-manual-p.txt'
+    sections_abs_path = os.path.join(head, sections_path)
+
+    sections = []
+    with open(sections_abs_path, 'r') as reader:
+        for line in reader.readlines():
+            if line != '\n':
+                sections.append(line.split(',')[3].split('\n')[0])
+
+    section_count = 0
+    with open(tempo_path, 'r') as reader:
+        for line in reader.readlines():
+
+            tempo_data = []
+            tempo = line.split(',')[0]
+            tempo_data.append(tempo)
+            matra = line.split(',')[1].split(' ')[1]
+            tempo_data.append(matra)
+            sama_interval = line.split(',')[2].split(' ')[1]
+            tempo_data.append(sama_interval)
+            matras_per_cycle = line.split(',')[3].split(' ')[1]
+            tempo_data.append(matras_per_cycle)
+            start_time = line.split(',')[4].split(' ')[1]
+            tempo_data.append(start_time)
+            duration = line.split(',')[5].split(' ')[1]
+            tempo_data.append(duration)
+
+            if 'NaN' in tempo_data:
+                return None
+
+            tempo_annotation[sections[section_count]] = {
+                'tempo': float(tempo) if '.' in tempo else int(tempo),
+                'matra_interval': float(matra) if '.' in matra else int(matra),
+                'sama_interval': float(sama_interval)
+                if '.' in sama_interval
+                else int(sama_interval),
+                'matras_per_cycle': float(matras_per_cycle)
+                if '.' in matras_per_cycle
+                else int(matras_per_cycle),
+                'start_time': float(start_time)
+                if '.' in start_time
+                else int(start_time),
+                'duration': float(duration) if '.' in duration else int(duration),
+            }
+
+            section_count += 1  # Go to next section
+
+    return tempo_annotation
+
+
+def load_sama(sama_path):
+    """Load sama
+
+    Args:
+        sama_path (str): Local path where the sama annotation is stored.
+            If `None`, returns None.
+
+    Returns:
+        SectionData: sama annotations
+
+    """
+    if sama_path is None:
+        return None
+
+    if not os.path.exists(sama_path):
+        raise IOError("sama_path {} does not exist".format(sama_path))
+
+    timestamps = []
+    sama_cycles = []
+    intervals = []
+    with open(sama_path, 'r') as reader:
+        for line in reader.readlines():
+            timestamps.append(float(line))
+
+        # Return None if sama file is empty
+        if not timestamps:
+            return None
+
+    for i in np.arange(1, len(timestamps)):
+        intervals.append([timestamps[i - 1], timestamps[i]])
+        sama_cycles.append('sama cycle ' + str(i))
+
+    return utils.SectionData(np.array(intervals), sama_cycles)
```

Comment:
I understand from [this paper](https://repositori.upf.edu/bitstream/handle/10230/32893/Srinivasamurthy_ISMIR2017_info.pdf?sequence=1&isAllowed=y) that `sama` is the downbeat, so we should use `BeatData` instead?

Reply:
Yes, we could! I was using `SectionData` because the annotations are "Sama cycles", instead of beat time-stamps and positions. What I don't clearly see is how to transform Sama annotations to `BeatData`. Should we set the `beat_positions` to 1 for every annotated time-stamp? I guess that would work because at the end of the day, the strings with `sama cycle X` wouldn't be needed.

Reply:
Is a good question, because the cycle information is similar to the `bar number` right? Another option is to include a `DownbeatData` type and include `bar_number`, but it feels redundant. Thinking of #291, actually you could extend the beat class to have bar bar number there no?

Reply:
I see! So with _extend the beat class_ you mean editting `BeatData` class so it has the `bar_number` attribute, or creating `DownbeatData` in my loader extending `BeatData` with the additional bar?

Reply:
I mean that after #336 is merged maybe we can add the `bar_number` more elegantly by inheriting the `BeatData` class and adding the attribute to the child, which would be implemented in the loader. But for now what you can do is to  implement `sama` using `BeatData` and we see if we include the bar number later

Reply:
Ok! Let's do it like that! I just pushed the updated version using `BeatData`.